docs/syntax/Syntax Notation.md

```diff
@@ -0,0 +1,255 @@
+# Syntax Notation
+
+The syntax file is a machine-readable file that describes the scanner and parser of a compiler frontend. This description is utilized to generate the frontend and the syntax tree.
+
+
+## Scanner Section
+
+The start of the section is denoted by a line containing only the text `%scanner`.
+
+### Token Rule
+
+Tokens are defined and referenced using angled brackets (`<` and `>`) surrounding their names. They are the sole scanner rules permitted in the parser. Tokens can have multiple productions separated by a pipe character (`|`). They can be marked with an asterisk (`*`) preceding the first angle bracket to indicate that when the token is matched, it is discarded.
+
+
+```c
+*
+<spaceToken>:
+        whitespace
+    ;
+```
+
+### Normal Rule
+
+Normal rules can have any number of productions separated by a pipe character (|) with each production containing one or more elements.
+
+
+```c
+whitespaceItem:
+        singleLineComment
+    |   multipleLineComment
+    |   whitespaceCharacter
+    |   lineBreak
+    ;
+```
+
+### Set Rule
+
+Set rules consist of a sequence of intervals separated by a comma (`,`). These intervals can either represent a single Unicode code point or a range of code points, separated by a hyphen (`-`).
```

Comment:
Aren't alternations with `|` also sets? Could we maybe just consistently use `|` to avoid unnecessarily adding a new operator uncommon in other notions